### PR TITLE
Employee Portal project demo 1

### PR DESCRIPTION
**Pull request by Abhirajsingh**
Review Project First demo

### CODE REVIEW

File: project/pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-jpa</artifactId>
+        </dependency>
```
Comment (shashanksheela) [inline project/pom.xml]: remove this dependency

File: project/pom.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    <dependency>
+        <groupId>org.hibernate</groupId>
+        <artifactId>hibernate-entitymanager</artifactId>
+        <version>5.2.12.Final</version>
+    </dependency>
+        <dependency>
+            <groupId>javax.persistence</groupId>
+            <artifactId>javax.persistence-api</artifactId>
+            <version>2.2</version>
+        </dependency>
```
Comment (shashanksheela) [inline project/pom.xml]: check even this,I dont thinks so this requires

File: project/src/main/java/com/zemoso/project/controller/DepartmentController.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    public ResponseEntity<Map<String, List<Map<String, Object>>>> getAllDeparmentOfCompany(){
+        Map<String,List<Map<String,Object>>> responseMap = new HashMap<>();
+        List<Map<String, Object>> mapList = new ArrayList<>();
+        try {
+            List<Department> departments = departmentService.getAllDepartment(CompanyUtil.getCompanyId());
+            departments.forEach(item -> {
+                Map<String, Object> departmentMap = null;
+                try {
+                    departmentMap = departmentMapper.getObjectMap(item);
+                } catch (MapperException e) {
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/controller/DepartmentController.java:50]: use object mapper

File: project/src/main/java/com/zemoso/project/controller/DepartmentController.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/departments")
+public class DepartmentController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DepartmentController.class);
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/controller/DepartmentController.java]: use @Self4j annotation for loggers

File: project/src/main/java/com/zemoso/project/controller/DepartmentController.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+        Map<String, Map<String, Object>> map = new HashMap<>();
+        try {
+            map.put("department", departmentMapper.getObjectMap(department) );
+        } catch (MapperException e) {
+            LOGGER.error(e.getMessage() , e);
+        }
+
+        return ResponseEntity.ok().body(map);
+
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/controller/DepartmentController.java]: remove this method as discussed

File: project/src/main/java/com/zemoso/project/controller/EmployeePortalController.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+       catch (NullPointerException e){
+           LOGGER.error(e.getMessage());
+       }
+       catch (DbException e){
+           LOGGER.error(e.getMessage() , e);
+       }
+
+
+       responseMap.put("employees" , mapList);
+       return ResponseEntity.ok().body(responseMap);
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/controller/EmployeePortalController.java]: do not use ok() method,use  HTTPSTATUS.OK

File: project/src/main/java/com/zemoso/project/controller/EmployeePortalController.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+    }
+    /**
+     * Rest controller to get selected employee
+     *
+     * @param employeeId
+     * @return
+     */
+    @RequestMapping(path = "/{employeeId}", method = RequestMethod.GET)
+    public ResponseEntity<Map<String, Object>> getProjectsById(@PathVariable Long employeeId) {
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/controller/EmployeePortalController.java]: change the method name

File: project/src/main/java/com/zemoso/project/controller/EmployeePortalController.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        try {
+            emap = employeeMapper.getObjectMap(employee);
+        } catch (MapperException e) {
+            LOGGER.error(e.getMessage() , e);
+        }
+        Map<String ,Object> responseMap = new HashMap<>();
+        responseMap.put(Constant.EMPLOYEE ,emap );
+
+        return ResponseEntity.ok().body(responseMap);
+    }
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/controller/EmployeePortalController.java]: exception handling should be improved

File: project/src/main/java/com/zemoso/project/model/Employee.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.Set;
+
+@Data
+@Entity
+public class Employee extends BaseEntityModel{
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/model/Employee.java]: use GenerationType.Sequence

File: project/src/main/java/com/zemoso/project/utils/EmployeeMapper.java [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+            }
+
+            if (map.containsKey(Constant.PROFILE_PIC))
+                employee.setProfilePic(map.get(Constant.PROFILE_PIC).toString());
+
+            employee.setCreatedById(CompanyUtil.getCompanyId());
+            employee.setLastUpdatedById(CompanyUtil.getCompanyId());
+            employee.setCompanyId(CompanyUtil.getCompanyId());
+
+            return employee;
```
Comment (shashanksheela) [inline project/src/main/java/com/zemoso/project/utils/EmployeeMapper.java:147]: use beanutils setproperty method

File: application/src/main/resources/db/migration/V1__User_data.sql [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+  OIDS=FALSE
+);
+ALTER TABLE public.department
+  OWNER TO postgres;
+
+INSERT INTO department VALUES (1, 1, 'Development');
+INSERT INTO department VALUES (2, 1, 'Testing');
+INSERT INTO department VALUES (3, 1, 'Human Resource');
+INSERT INTO department VALUES (4, 1, 'Accountancy');
+INSERT INTO department VALUES (5, 1, 'Quality Assurance');
```
Comment (shashanksheela) [inline application/src/main/resources/db/migration/V1__User_data.sql:35]: extend basentity from all tables

File: ui/app/models/skill.js [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import DS from 'ember-data';
+
+export default DS.Model.extend({
+employee: DS.hasMany('employee'),
```
Comment (vijaysai88zemoso) [inline ui/app/models/skill.js]: not required

**Comment by Abhirajsingh** [comment]
#

**Comment by Abhirajsingh** [comment]
Sprint 1 pull request

**Pull request closed without merging**